runtime/domain.c

```diff
@@ -1275,6 +1263,15 @@ static void domain_terminate()
     acknowledge_all_pending_interrupts();
   }
 
+  /* release domain blocked waiting to join this domain */
+  check_err("domain_terminate_waiter_lock",
```

Comment:
Unfortunately accessing `ml_values` here is unsafe. We are no longer a domain participating in STW sections and so a GC may race to update `ml_values->mutex`, `ml_values->condition`, etc.

We will need another approach...